File: Problem_02/AC_solution.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int n;
    string a, b;
    cin >> n >> a >> b;
    int ai, bi;
    bool turn = true;
    cout << a.size() << endl;
    for (int i = 0; i < n; i++) {
        ai = a[i] - '0';
        bi = b[i] - '0';
        if (ai < bi) turn = !turn;
        else if (ai == bi) break;
    }
    cout << ((turn) ? "Luffy" : "Zoro");

    return 0;
}

File: Problem_02/AC_solution.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String a = scanner.next();
        String b = scanner.next();
        int ai, bi;
        boolean turn = true;

        for (int i = 0; i < n; i++) {
            ai = a.charAt(i) - '0';
            bi = b.charAt(i) - '0';
            if (ai < bi) {
                turn = !turn;
            } else if (ai == bi) {
                break;
            }
        }

        System.out.println(turn ? "Luffy" : "Zoro");
    }
}
